Code/Tools/AssetProcessor/native/ui/BuilderInfoMetricsModel.cpp

```diff
@@ -0,0 +1,266 @@
+#include <AssetBuilderSDK/AssetBuilderSDK.h>
+#include <QTime>
+
+namespace AssetProcessor
+{
+    QString DurationToQString(AZ::s64 durationInMs)
+    {
+        const AZ::s64 dayInMs = 86400000;
+        if (durationInMs < 0)
+        {
+            return "";
+        }
+
+        AZ::s64 dayCount = durationInMs / dayInMs;
+        QTime duration = QTime::fromMSecsSinceStartOfDay(durationInMs % dayInMs);
+        if (dayCount > 0)
+        {
+            return duration.toString("zzz' ms, 'ss' sec, 'mm' min, 'hh' hr, %1 day'").arg(dayCount);
+        }
+        else
+        {
+            if (!duration.isValid())
+            {
+                return "";
+            }
+            if (duration.hour() > 0)
+            {
+                return duration.toString("zzz' ms, 'ss' sec, 'mm' min, 'hh' hr'");
+            }
+            if (duration.minute() > 0)
+            {
+                return duration.toString("zzz' ms, 'ss' sec, 'mm' min'");
+            }
+            if (duration.second() > 0)
+            {
+                return duration.toString("zzz' ms, 'ss' sec'");
+            }
+            return duration.toString("zzz' ms'");
+        }
+    }
+
+    BuilderInfoMetricsModel::BuilderInfoMetricsModel(BuilderData* builderData, QObject* parent)
+        : QAbstractItemModel(parent)
+        , m_data(builderData)
+    {
+    }
+
+    void BuilderInfoMetricsModel::Reset()
+    {
+        beginResetModel();
+        m_data->Reset();
+        endResetModel();
+    }
+
+    void BuilderInfoMetricsModel::OnBuilderSelectionChanged(const AssetBuilderSDK::AssetBuilderDesc& builder)
+    {
+        beginResetModel();
+        
+        if (m_data->m_builderGuidToIndex.contains(builder.m_busId))
+        {
+            m_data->m_currentSelectedBuilderIndex = m_data->m_builderGuidToIndex[builder.m_busId];
+            m_data->m_root->SetBuilderChild(m_data->m_singleBuilderMetrics[m_data->m_currentSelectedBuilderIndex]);
+        }
+        else
+        {
+            AZ_Warning(
+                "Asset Processor",
+                false,
+                "BuilderInfoMetricsModel cannot find the GUID of the builder selected by the user (%s) in itself. No metrics will be "
+                "shown in the builder tab.",
+                builder.m_busId.ToString<AZStd::string>().c_str());
+            m_data->m_currentSelectedBuilderIndex = aznumeric_cast<int>(BuilderData::BuilderSelection::Invalid);
+        }
+        
+        endResetModel();
+    }
+
+    QModelIndex BuilderInfoMetricsModel::index(int row, int column, const QModelIndex& parent) const
+    {
+        if (!hasIndex(row, column, parent))
+        {
+            return QModelIndex();
+        }
+
+        BuilderDataItem* parentItem = nullptr;
+
+        if (!parent.isValid())
+        {
+            parentItem = m_data->m_root.get();
+        }
+        else
+        {
+            parentItem = static_cast<BuilderDataItem*>(parent.internalPointer());
+        }
```

Comment:
I would probably combine this into a ternary
```suggestion
        BuilderDataItem* const parentItem = parent.isValid() ? static_cast<BuilderDataItem*>(parent.internalPointer()) : m_data->m_root.get(); 
```